app/assets/javascripts/router.js

```diff
@@ -179,8 +187,20 @@ OSM.Router = function (map, rts) {
   };
 
   router.load = function () {
-    var loadState = currentRoute.run("load", currentPath);
+    var loadState = loadOrReload();
```

Comment:
What's the reason for putting this logic in a nested function if this is the only user?

Reply:
Are you asking why it's nested or why it's defined when there's only one caller?

The short answer is to initialize var loadState when it's declared.

Reply:
I'm asking why put the logic in a function rather than just having it inline.

Are you saying that function is only called once rather than each time the load function is invoked?

Reply:
Of course loadOrReload is called as many times as router.load.

Reply:
Changed it to extra variable instead of extra function.